File: LibraryProject/ui/src/app/share/entity/book.ts
export class book{

  private _id: any;
  private _name: string;
  private _author: string;
  private _year: string;
  private _isbn: string;


  constructor(bookID: any, author: string,name: string, year: string, isbn: string) {
    this._id = bookID;
    this._author = author;
    this._name = name;
    this._year = year;
    this._isbn = isbn;
  }


  set bookID(value: any) {
    this._id = value;
  }

  set name(value: string) {
    this._name = value;
  }

  set author(value: string) {
    this._author = value;
  }

  set year(value: string) {
    this._year = value;
  }

  set isbn(value: string) {
    this._isbn = value;
  }

  get bookID(): any {
    return this._id;
  }

  get name(): string {
    return this._name;
  }

  get author(): string {
    return this._author;
  }

  get year(): string {
    return this._year;
  }

  get isbn(): string {
    return this._isbn;
  }
}

File: LibraryProject/Springboot/src/main/java/com/beriani/library/model/Book.java
package com.beriani.library.model;

import javax.persistence.*;
import java.util.List;

@Entity(name = "book")
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int bookID;

    private String author;
    private String name;
    private int year;
    private int isbn;

    @ManyToMany(mappedBy = "books")
    private List<User> users;

    public Book(){}

    public int getBookID() {
        return bookID;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public void setIsbn(int isbn) {
        this.isbn = isbn;
    }



    public String getAuthor() {
        return author;
    }

    public String getName() {
        return name;
    }

    public int getYear() {
        return year;
    }

    public int getIsbn() {
        return isbn;
    }
}
